### PR TITLE
HDDS-8400. Expose rocksdb last sequence number through metrics

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/HDDS-8400

## How was this patch tested?

captured metrics from service /jmx endpoint 

image

image

### CODE REVIEW

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RocksDBStoreMetrics.java [hunk 1]
```diff
@@ -105,6 +105,8 @@ public class RocksDBStoreMetrics implements MetricsSource {
   private static final String NUM_FILES_AT_LEVEL = "num_files_at_level";
   private static final String SIZE_AT_LEVEL = "size_at_level";
 
+  private static final String LAST_SEQUENCE_NUMBER = "last_sequence_number";
```
Comment (krishnaasawa1) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RocksDBStoreMetrics.java:108]: Looks straightforward change introducing  LAST_SEQUENCE_NUMBER metrics. On similar lines as HistogramData,TickerTypeData,PropertyData we have introduced getLatestSequenceNumber and updating metrics with calling  getLatestSequenceNumber via getMetrics.

Comment (krishnaasawa1) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RocksDBStoreMetrics.java:108]: @mohammadjkhan @devmadhuu Please Review as metrics related

**Review by krishnaasawa1 — APPROVED** [review]

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RocksDBStoreMetrics.java (hunk 1 above)
Comment (devmadhuu) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RocksDBStoreMetrics.java:108]: Can we also expose wal_ttl in metrics and wal_size ?

Comment (ChenSammi) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RocksDBStoreMetrics.java:108]: @devmadhuu ， wal_ttl and wal_size value are statically configured value. They can be retrieved from the rocksdb MANIFEST file.

**Review by ayushtkn — COMMENTED** [review]
Prod code LGTM, is there any scope to extend a test as well, may be just verifying this number gets populated?

**Review by ayushtkn — APPROVED** [review]
LGTM
Thanx Sammi for creating a followup for adding tests for metrics

**Comment by ChenSammi** [comment]
Thanks @krishnaasawa1 @devmadhuu @ayushtkn for the code review.

**Comment by adoroszlai** [comment]
> there is no metrics related integration test in Ozone now

We don't need separate tests for metrics.  Assertions about metrics should be added to existing tests.

**Pull request merged**